riscv/simif.h

```diff
@@ -11,6 +11,7 @@ class simif_t
 public:
   // should return NULL for MMIO addresses
   virtual char* addr_to_mem(reg_t addr) = 0;
+  virtual bool reservable(reg_t addr) { return addr_to_mem(addr); };
```

Comment:
The param should be named `paddr`

Reply:
To maintain consistency, I've changed all of the simif_t member functions to use `paddr` instead of `addr`.